aws_topology/stackstate_checks/aws_topology/resources/api_gateway.py

```diff
@@ -25,135 +35,206 @@ def create_method_arn(region=None, account_id=None, resource_id=None, **kwargs):
-                                )
-                            )
-
-                            method_integration_uri = (
-                                replace_stage_variables(method_data["methodIntegration"]["uri"], stage["variables"])
-                                if "variables" in stage
-                                else method_data["methodIntegration"]["uri"]
-                            )
-                            method_data["methodIntegration"]["uri"] = method_integration_uri
-
-                            integration_arn = None
-                            if method_data["methodIntegration"]["type"] == "AWS_PROXY":
-                                integration_arn = method_integration_uri[
-                                    method_integration_uri.rfind("arn") : method_integration_uri.find("/invocations")
-                                ]
-                            elif re.match("arn:aws:apigateway:.+:sqs:path/.+", method_integration_uri):
-                                queue_arn = arn(
-                                    resource="sqs",
-                                    region=method_integration_uri.split(":", 4)[3],
-                                    account_id=method_integration_uri.split("/", 2)[1],
-                                    resource_id=method_integration_uri.rsplit("/", 1)[-1],
-                                )
-                                integration_arn = queue_arn
-
-                            self.emit_component(method_arn, "aws.apigateway.method", method_data)
-                            self.emit_relation(resource_arn, method_arn, "uses service", {})
-
-                            if integration_arn:
-                                self.emit_relation(method_arn, integration_arn, "uses service", {})
-
-                            # Creates a dummy service component that is connected to the api gateway method
-                            # this dummy service component will merge with a real trace service
-                            if method_data["methodIntegration"]["type"] == "HTTP_PROXY":
-                                parsed_uri = urlparse(method_data["methodIntegration"]["uri"])
-                                service_integration_urn = "urn:service:/{0}".format(parsed_uri.hostname)
-
-                                self.emit_component(
-                                    service_integration_urn, "aws.apigateway.method.http.integration", {}
-                                )
-                                self.emit_relation(method_arn, service_integration_urn, "uses service", {})
+                )
+            )
+            self.emit_component(resource_arn, "aws.apigateway.resource", resource_data)
+            self.emit_relation(stage_arn, resource_arn, "uses service", {})
+
+            for method_id in resource.resourceMethods.keys():
+                self.process_resource_method(
+                    data["resourceMethods"][method_id],
```

Comment:
Use the schematic here

Reply:
I need the original json object, schematics would lose all values that I haven't defined. Line 190 includes the schematics version